tss-esapi/tests/integration_tests/abstraction_tests/credential_tests.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2025 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use tss_esapi::{
+    abstraction::{ak, ek, AsymmetricAlgorithmSelection},
+    attributes::SessionAttributesBuilder,
+    constants::SessionType,
+    handles::AuthHandle,
+    interface_types::{
+        algorithm::{HashingAlgorithm, SignatureSchemeAlgorithm},
+        ecc::EccCurve,
+        session_handles::PolicySession,
+    },
+    structures::{Digest, SymmetricDefinition},
+    utils,
+};
+
+use elliptic_curve::PublicKey;
+
+use crate::common::create_ctx_without_session;
+
+#[test]
+fn test_credential() {
+    let mut context = create_ctx_without_session();
+
+    let ek_ecc = ek::create_ek_object(
+        &mut context,
+        AsymmetricAlgorithmSelection::Ecc(EccCurve::NistP256),
+        None,
+    )
+    .unwrap();
+
+    let (ek_pub, _, _) = context.read_public(ek_ecc).unwrap();
+
+    let ak_res = ak::create_ak(
+        &mut context,
+        ek_ecc,
+        HashingAlgorithm::Sha384,
+        AsymmetricAlgorithmSelection::Ecc(EccCurve::NistP384),
+        SignatureSchemeAlgorithm::EcDsa,
+        None,
+        None,
+    )
+    .unwrap();
+
+    let ak_ecc = ak::load_ak(
+        &mut context,
+        ek_ecc,
+        None,
+        ak_res.out_private,
+        ak_res.out_public,
+    )
+    .unwrap();
+
+    let (_, key_name, _) = context.read_public(ak_ecc).unwrap();
+    let cred = vec![1, 2, 3, 4, 5];
+    let expected = Digest::try_from(vec![1, 2, 3, 4, 5]).unwrap();
+
+    let (credential_blob, secret) = utils::make_credential_ecc::<_, sha2::Sha256, aes::Aes128>(
```

Comment:
`sha2::Sha256, aes::Aes128` here codes for `EKHash` and `EkCipher`.
Those should be read from the template of the EK ideally.

Although in realily, the template would have been dropped already and we're only working with a PEM encoded public key, and there should be some kind of default value.
https://github.com/tpm2-software/tpm2-tools/blob/master/tools/tpm2_makecredential.c#L340

Reply:
Is your question about this test in particular, or about the interface of `make_credential_ecc`, and whether we can deduce the type params from the inputs?

I think generally you should be able to deduce the hash and the cipher for the EK if you know the nature of the public key, for example by doing the reverse of the mapping done here: https://github.com/parallaxsecond/rust-tss-esapi/blob/main/tss-esapi/src/abstraction/ek.rs

Reply:
No that was an API question.

I know how to get the parameters from a Public, but I don't expect the public or its template to always available.